### DISCUSSION TITLE
Prometheus metric question

### DISCUSSION THREAD

**Discussion by jiangxinlingdu** [rabbitmq/rabbitmq-server#2781] [Other] [answered]
i use rabbitmq 3.8.5 
![image](https://user-images.githubusercontent.com/6373972/106577710-a80ffb80-6579-11eb-9202-346538154573.png)

![image](https://user-images.githubusercontent.com/6373972/106577763-b78f4480-6579-11eb-9836-a1a190662a67.png)
rabbitmq prometheus metrics  rabbitmq_queue_messages_published_total  is counter type but have a little problem，rabbitmq_queue_messages_published_total   can down 
![image](https://user-images.githubusercontent.com/6373972/106577978-fc1ae000-6579-11eb-9d38-c4ca7681561b.png)
This question bothers me a little……

i use http api, also have are problems
![image](https://user-images.githubusercontent.com/6373972/106578350-63d12b00-657a-11eb-9774-ea7717d2c93b.png)

**Comment by michaelklishin** [comment]
I will convert this issue to a GitHub discussion. Currently GitHub will [automatically close and lock the issue](https://github.com/github/feedback/discussions/3197) even though your question will be transferred and responded to elsewhere. This is to let you know that we do not intend to ignore this but this is how the current GitHub conversion mechanism makes it seem for the users :(

**Comment by michaelklishin** [comment]
I'm afraid I do not understand what the "problem(s)" is/are. Metrics are emitted [periodically](https://www.rabbitmq.com/monitoring.html#monitoring-frequency). In case of Prometheus, they will also be [aggregated by default](https://www.rabbitmq.com/prometheus.html#metric-aggregation) but HTTP API will always serve per-object metrics
(there is no way to switch to totals only). It is not clear what your charts demonstrate.

There are multiple metrics for "published messages": published on a channel, routed to a queue (ingress for a queue), totals.
Published messages [can be unroutable](https://www.rabbitmq.com/publishers.html#unroutable), there are separate metrics for those (returned and dropped).

There isn't enough information for us

**Comment by michaelklishin** [comment]
With some help from a Mandarin-speaking team member we now think that the surprising part is that the rate can stay stable but the total can decrease. One scenario where this will happen is when a channel (or connection) is closed: then all of their metric (samples) will be deleted. In other words, these metrics are tied to client-controlled entities with a finite lifecycle, not the lifecycle of the node or cluster. There are no counters over the lifecycle of the cluster.

**Reply by jiangxinlingdu** [reply]
yeah, that's the plan

**Comment by gerhard** [comment, marked as answer]
We have confirmed that this is an issue, thanks for reporting it @jiangxinlingdu https://github.com/rabbitmq/rabbitmq-server/issues/2783
